cs/src/Management/TunnelManagementClient.cs

```diff
@@ -1096,17 +977,20 @@ public void Dispose()
             Requires.NotNull(endpoint, nameof(endpoint));
             Requires.NotNullOrEmpty(endpoint.HostId!, nameof(TunnelEndpoint.HostId));
 
-            var path = $"{EndpointsApiSubPath}/{endpoint.HostId}/{endpoint.ConnectionMode}";
+            var path = $"{EndpointsApiSubPath}/{endpoint.Id}";
+            var query = GetApiQuery();
+            query += "&connectionMode=" + endpoint.ConnectionMode;
```

Comment:
Does `endpoint.ConnectionMode` need to be URI encoded?

Reply:
No, there are only 2 values, tunnelRelay and localNetwork